nucleus/payara-modules/healthcheck-core/src/main/java/fish/payara/nucleus/healthcheck/admin/SetHealthCheckConfiguration.java

```diff
@@ -0,0 +1,207 @@
+    private int historicalTraceStoreSize;
+
+    @Param(name = "trace-store-timeout", optional = true)
+    private String historicalTraceStoreTimeout;
+
+    @Override
+    public void execute(AdminCommandContext context) {
+        targetConfig = targetUtil.getConfig(target);
+        HealthCheckServiceConfiguration config = targetConfig.getExtensionByType(HealthCheckServiceConfiguration.class);
+        if (config != null) {
+            updateConfig(config, context);
+        }
+        if (dynamic && (!server.isDas() || targetConfig.isDas())) {
+            configureDynamically();
+        }
+        updateLogNotifier(context);
+    }
+
+    private static ActionReport initActionReport(AdminCommandContext context) {
+        final ActionReport report = context.getActionReport();
+        if (report.getExtraProperties() == null) {
+            report.setExtraProperties(new Properties());
+        }
+        return report;
+    }
+
+    private void updateConfig(HealthCheckServiceConfiguration config, AdminCommandContext context) {
+        final ActionReport report = initActionReport(context);
+        try {
+            ConfigSupport.apply(proxy -> {
+                    proxy.enabled(String.valueOf(enabled));
+                    proxy.setHistoricalTraceStoreSize(String.valueOf(historicalTraceStoreSize));
+                    if (historicalTraceEnabled != null) {
+                        proxy.setHistoricalTraceEnabled(historicalTraceEnabled.toString());
+                    }
+                    if (historicalTraceStoreTimeout != null) {
+                        proxy.setHistoricalTraceStoreTimeout(historicalTraceStoreTimeout.toString());
+                    }
+                    report.setActionExitCode(ActionReport.ExitCode.SUCCESS);
+                    return proxy;
+                }, config);
+        } catch (TransactionFailure ex) {
+            logger.log(Level.WARNING, "Exception during command ", ex);
+            report.setMessage(ex.getCause() != null ? ex.getCause().getMessage() : ex.getMessage());
+            report.setActionExitCode(ActionReport.ExitCode.FAILURE);
+            return;
+        }
+    }
+
+    private void updateLogNotifier(AdminCommandContext context) {
+        CommandRunner runner = serviceLocator.getService(CommandRunner.class);
+        ActionReport subReport = context.getActionReport().addSubActionsReport();
+        CommandRunner.CommandInvocation inv = runner.getCommandInvocation("set-healthcheck-service-notifier-configuration", subReport, context.getSubject());
+        ParameterMap params = new ParameterMap();
+        params.add("target", target);
+        params.add("dynamic", String.valueOf(dynamic));
+        params.add("enabled", String.valueOf(enabled));
+        params.add("notifier", NotifierType.LOG.name().toLowerCase());
+        // noisy will default to the notifier's config when not set so we do not set it as this is what we want
```

Comment:
I intentionally removed the preset of `noisy` done in the old command as the called service does just have that exact logic if the parameter is not supplied so I think its better to not supply it and have the called command take care of providing the default.